src/Mvc/Mvc.ViewFeatures/src/SkipStatusCodePagesAttribute.cs

```diff
@@ -11,15 +11,23 @@ namespace Microsoft.AspNetCore.Mvc
     /// A filter that prevents execution of the StatusCodePages middleware.
     /// </summary>
     [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
-    public class SkipStatusCodePagesAttribute : Attribute, IResourceFilter
+    public class SkipStatusCodePagesAttribute : Attribute, IAlwaysRunResultFilter, IOrderedFilter
```

Comment:
While I think this change makes sense, this does 

a) is a breaking change
b) does not work with endpoint routing

We should leave the `IResourceFilter` on there since it does not hurt to invoke the filter in multiple stages and allows it to be non-breaking. In addition, as of 3.0, Auth happens as part of the middleware pipeline, so this change really doesn't help there. If you'd like to solve it for the latter, one solution would be to:

a) Introduce an `ISkipStatusCodePagesMetadata`
b) Use the presence of the metadata to skip the status code page if you're doing endpoint routing. Here's a similar pattern: https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/CORS/src/Infrastructure/CorsMiddleware.cs#L140-L159

Reply:
If you want to consider skipping solving this for endpoint metadata, let me know and I can file a separate issue to track it. cc @JamesNK \ @rynowak \ @Tratcher